packages/web/utils/cache.ts

```diff
@@ -15,28 +31,44 @@ const isTestEnv = process.env.NODE_ENV === "test";
  *  NOTE: Uses an LRUCache in test environment to avoid hitting the remote resource as Vercel KV createClient secret env vars are not available in GH action.
  *
  *  WARNING: Only available in dev/prod node runtime (not browser).
+ *
+ *  Falls back to in-memory cache if the remote client is not able to be created.
  */
 export class RemoteCache implements Cache {
   protected kvStore: VercelKV | null = null;
 
-  protected testEnvKvStore = isTestEnv
-    ? new LRUCache<string, CacheEntry>({ max: 50 })
-    : null;
+  protected fallbackCache: Cache | null = null;
 
   name = "RemoteCache";
 
   constructor() {
     if (!isTestEnv) {
-      this.kvStore = createClient({
-        url: process.env.TWITTER_KV_STORE_REST_API_URL!,
-        token: process.env.TWITTER_KV_STORE_REST_API_TOKEN!,
-      });
+      try {
+        this.kvStore = createClient({
+          url: process.env.TWITTER_KV_STORE_REST_API_URL!,
+          token: process.env.TWITTER_KV_STORE_REST_API_TOKEN!,
+        });
+      } catch (e) {
+        console.error(
+          "Failed to create RemoteCache client. Falling back to in-memory cache.."
+        );
+        this.fallbackCache = new LRUCache<string, CacheEntry>(
+          DEFAULT_LRU_OPTIONS
+        );
+      }
+    } else {
+      console.warn(
+        "RemoteCache is not available in test environment. Falling back to in-memory cache.."
+      );
+      this.fallbackCache = new LRUCache<string, CacheEntry>(
+        DEFAULT_LRU_OPTIONS
+      );
     }
   }
 
   async get<T>(key: string) {
```

Comment:
@jonator I believe we have to prefix keys based on the environment. IIRC, Vercel KV does not differ between envs so we could have some data conflicts

Reply:
WDYM by environment? Aren't shared keys a key value prop with the remote cache?

Reply:
I mean envs like production, testnet, and edgenet

Reply:
Ah that's true. What are all of the environments?
* Mainnet
* Testnet
* Edgenet (any other chain env)

What should we use to derive these environment in your opinion?

Reply:
I agree that other environments like testnet and edgenet can utilize in-memory caching. We can introduce a check for environment variables. If they're missing, we should default to the in-memory cache. Additionally, it would be wise to set up a KV for stage/previews so we can QA the changes

Reply:
OK. I'll check for chain IDs set in `process.env.NEXT_PUBLIC_OSMOSIS_CHAIN_ID_OVERWRITE`, as those are not expected to change for mainnet and testnet. Should we check against hardcoded values or simply prefix with that variable value?

Then, if it's not set we could divert to in-memory cache.

You're saying we should have a separate KV store for stage vs prod?

Reply:
> You're saying we should have a separate KV store for stage vs prod?

Yes, we should distinguish between them by using different KV urls, particularly because they access the same underlying datasets, such as the asset list and chain list